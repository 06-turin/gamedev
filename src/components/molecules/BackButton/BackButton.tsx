import React, { FC } from 'react';
import { GDButton } from 'components/atoms/GDButton/GDButton';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

export const BackButton: FC = () => {
  const { t } = useTranslation();
  const history = useHistory();

  return (
    <GDButton
      title={t('back')}
      styleOption="secondary"
      size="l"
      onClick={() => history.back()}
    />
  );
};
