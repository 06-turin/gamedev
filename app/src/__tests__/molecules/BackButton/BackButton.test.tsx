import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { I18nextProvider } from 'react-i18next';
import i18n from 'i18nForTests';
import { BackButton } from 'components/molecules/BackButton/BackButton';

describe('BackButton', () => {
  it('check i18n setup', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <BackButton />
        </I18nextProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
