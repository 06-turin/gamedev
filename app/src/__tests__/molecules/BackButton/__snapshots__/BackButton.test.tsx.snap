// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton check i18n setup 1`] = `
<button
  className="btn btn-secondary size_l"
  onClick={[Function]}
  type="button"
>
  back
</button>
`;
